### PR TITLE
BaSP-TG-72: FIX Project validations

### PR DESCRIPTION
**Pull request by fede230298**
1) End Date now must be higher than Start Date to be valid
2) Name, Client Name, Description no longer allows it to be completed with only blank spaces, only numbers or only special characters. The Empty spaces at the edges do not count to the minimum of 3 characters
3) Rate now must be a positive number ( greater than 0: number > 0). No longer allows negative numbers

### CODE REVIEW

**Review by ignacioaristo — APPROVED** [review]

File: src/validations/projects.js [hunk 1]
```diff
@@ -4,14 +4,17 @@ const validateCreation = (req, res, next) => {
...
-    name: Joi.string().min(3).required(),
-    description: Joi.string().min(3).required(),
+    name: Joi.string().required().trim().regex(/^(?=.*[a-zA-Z].*)([\w\s\W]+)$/)
+      .min(3),
+    description: Joi.string().required().trim().regex(/^(?=.*[a-zA-Z].*)([\w\s\W]+)$/)
+      .min(3),
     startDate: Joi.date().required(),
-    endDate: Joi.date().required(),
-    clientName: Joi.string().min(3).required(),
+    endDate: Joi.date().greater(Joi.ref('startDate')).required(),
```
Comment (JuliMonte5) [inline src/validations/projects.js:15]: Maybe shouldn't be required, for example if it's an ongoing project

Comment (fede230298) [reply src/validations/projects.js:15]: I think all projects should have a deadline, and if it needs to be changed, you can edit the project.

Comment (JuliMonte5) [reply src/validations/projects.js:15]: Right, seen as a deadline has more sense

**Review by JuliMonte5 — APPROVED** [review]

**Review by sofihoj — APPROVED** [review]
Great work Fede!

**Pull request merged**